crates/language-server/src/backend/get_core.rs

```diff
@@ -0,0 +1,85 @@
+use camino::Utf8PathBuf;
+use common::{
+    input::{IngotKind, Version},
+    InputDb, InputFile, InputIngot,
+};
+use tracing::info;
+
+#[derive(rust_embed::RustEmbed)]
+#[folder = "../../library/core"]
+pub struct CoreLib;
+
+#[salsa::tracked]
+pub fn get_core_ingot(db: &dyn InputDb) -> InputIngot {
+    InputIngot::new(
+        db,
+        "core", // this should be handled by the driver crate default behavior
+        IngotKind::Core,
+        Version::new(0, 0, 1),
+        Default::default(),
+    )
+}
+
+pub fn init_core_ingot(db: &mut dyn InputDb) -> InputIngot {
```

Comment:
This would be better implemented in the driver of course, I just kept it here for this sketch